### PR TITLE
fix: media image assets [WPB-5848]

### PR DESCRIPTION
**Pull request by Garzas**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Added flow to listen already downloaded conversation images, because asynchronous fetch of image with paging library is causing out of memory error where we can't control load more behaviour.

### CODE REVIEW

**Review by saleniuk — APPROVED** [review]

File: logic/src/androidMain/kotlin/com/wire/kalium/logic/feature/asset/GetPaginatedFlowOfAssetImageMessageByConversationIdUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import com.wire.kalium.util.KaliumDispatcher
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flowOn
+
+/**
+ * This use case will observe and return a flow of paginated image asset messages for a given conversation.
+ * @see PagingData
+ * @see AssetMessage
+ */
+class GetPaginatedFlowOfAssetImageMessageByConversationIdUseCase internal constructor(
```
Comment (borichellow) [inline logic/src/androidMain/kotlin/com/wire/kalium/logic/feature/asset/GetPaginatedFlowOfAssetImageMessageByConversationIdUseCase.kt:34]: IMO something like `ObservePaginatedAssetImageMessages` would be smaller but clear enough :)

Comment (borichellow) [reply logic/src/androidMain/kotlin/com/wire/kalium/logic/feature/asset/GetPaginatedFlowOfAssetImageMessageByConversationIdUseCase.kt:34]: And why without interface?

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/MessageRepositoryExtensions.kt [hunk 2]
```diff
@@ -50,6 +52,12 @@ internal interface MessageRepositoryExtensions {
         pagingConfig: PagingConfig,
         startingOffset: Long
     ): Flow<PagingData<Message.Standalone>>
+
+    suspend fun getPaginatedMessageAssetImageByConversationId(
```
Comment (borichellow) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/MessageRepositoryExtensions.kt:56]: as far as it returns `Flow` - `observePaginatedMessageAssetImageByConversationId` fits better, imo

**Review by borichellow — APPROVED** [review]
Looks good, just few small suggestions :)

**Pull request merged**